### ISSUE TITLE
incomplete state for google_compute_instance_group_manager with count > 1

### ISSUE THREAD

**Issue by lobeck** [hashicorp/terraform#11828]
When using the `count` attribute in conjunction with `google_compute_instance_group_manager`, only the state of the first instance will be persisted.

We have a configuration like the following, where `var.regions` is a list of multiple regions and `var.zones` is a map with a region => list(zone) mapping.

```hcl
resource "google_compute_instance_group_manager" "consul" {
  count = "2"
  zone = "${element(var.regions, count.index)}-${element(var.zones[element(var.regions, count.index)], (count.index) / length(var.regions))}"
  name = "${var.name}-${element(var.regions, count.index)}"
  update_strategy    = "NONE"

  target_size  = "${var.count}"

  base_instance_name = "${var.name}-${element(var.regions, count.index)}"

  instance_template  = "${element(google_compute_instance_template.consul.*.self_link, count.index)}"

}
```

On initial creation, all resources are properly created as expected, but only the first one will be saved to state, so additional calls will cause recreation of the already existing instances and therefore fail.

This is the recorded state, where `count="2"` and two were created.

```json
"google_compute_instance_group_manager.consul": {
    "type": "google_compute_instance_group_manager",
    "depends_on": [
        "google_compute_instance_template.consul"
    ],
    "primary": {
        "id": "consul-europe-west1",
        "attributes": {
            "base_instance_name": "consul-europe-west1",
            "description": "",
            "fingerprint": "redacted",
            "id": "consul-europe-west1",
            "instance_group": "https://www.googleapis.com/compute/v1/projects/redacted/zones/europe-west1-d/instanceGroups/consul-europe-west1",
            "instance_template": "https://www.googleapis.com/compute/v1/projects/redacted/global/instanceTemplates/consul-europe-west1",
            "name": "consul-europe-west1",
            "named_port.#": "0",
            "project": "redacted",
            "self_link": "https://www.googleapis.com/compute/v1/projects/redacted/zones/europe-west1-d/instanceGroupManagers/consul-europe-west1",
            "target_pools.#": "0",
            "target_size": "3",
            "update_strategy": "NONE",
            "zone": "europe-west1-d"
        },
        "meta": {},
        "tainted": false
    },
    "deposed": [],
    "provider": ""
},
```

The initial `terraform plan`:

```
+ module.infrastructure.consul.google_compute_instance_group_manager.consul.0
    base_instance_name: "consul-europe-west1"
    fingerprint:        "<computed>"
    instance_group:     "<computed>"
    instance_template:  "https://www.googleapis.com/compute/v1/projects/redacted/global/instanceTemplates/consul-europe-west1"
    name:               "consul-europe-west1"
    project:            "<computed>"
    self_link:          "<computed>"
    target_size:        "3"
    update_strategy:    "NONE"
    zone:               "europe-west1-d"

+ module.infrastructure.consul.google_compute_instance_group_manager.consul.1
    base_instance_name: "consul-us-east1"
    fingerprint:        "<computed>"
    instance_group:     "<computed>"
    instance_template:  "https://www.googleapis.com/compute/v1/projects/redacted/global/instanceTemplates/consul-us-east1"
    name:               "consul-us-east1"
    project:            "<computed>"
    self_link:          "<computed>"
    target_size:        "3"
    update_strategy:    "NONE"
    zone:               "us-east1-d"


Plan: 2 to add, 0 to change, 0 to destroy.
```

The `terraform apply`:

```
module.infrastructure.consul.google_compute_instance_group_manager.consul.1: Creating...
  base_instance_name: "" => "consul-us-east1"
  fingerprint:        "" => "<computed>"
  instance_group:     "" => "<computed>"
  instance_template:  "" => "https://www.googleapis.com/compute/v1/projects/redacted/global/instanceTemplates/consul-us-east1"
  name:               "" => "consul-us-east1"
  project:            "" => "<computed>"
  self_link:          "" => "<computed>"
  target_size:        "" => "3"
  update_strategy:    "" => "NONE"
  zone:               "" => "us-east1-d"
module.infrastructure.consul.google_compute_instance_group_manager.consul.0: Creating...
  base_instance_name: "" => "consul-europe-west1"
  fingerprint:        "" => "<computed>"
  instance_group:     "" => "<computed>"
  instance_template:  "" => "https://www.googleapis.com/compute/v1/projects/redacted/global/instanceTemplates/consul-europe-west1"
  name:               "" => "consul-europe-west1"
  project:            "" => "<computed>"
  self_link:          "" => "<computed>"
  target_size:        "" => "3"
  update_strategy:    "" => "NONE"
  zone:               "" => "europe-west1-d"
module.infrastructure.consul.google_compute_instance_group_manager.consul.1: Still creating... (10s elapsed)
module.infrastructure.consul.google_compute_instance_group_manager.consul.0: Still creating... (10s elapsed)
module.infrastructure.consul.google_compute_instance_group_manager.consul.0: Creation complete
module.infrastructure.consul.google_compute_instance_group_manager.consul.1: Creation complete
```

A `terraform plan` now:

```
+ module.infrastructure.consul.google_compute_instance_group_manager.consul.1
    base_instance_name: "consul-us-east1"
    fingerprint:        "<computed>"
    instance_group:     "<computed>"
    instance_template:  "https://www.googleapis.com/compute/v1/projects/redacted/global/instanceTemplates/consul-us-east1"
    name:               "consul-us-east1"
    project:            "<computed>"
    self_link:          "<computed>"
    target_size:        "3"
    update_strategy:    "NONE"
    zone:               "us-east1-d"
```

`Terraform v0.8.5` & `Terragrunt v0.9.9`

**Comment by lobeck** [comment]
Just noticed, that there is `v0.8.6`, which seems to contain a fix, will have a look (if i find that release somewhere).

**Comment by lobeck** [comment]
Fixed with 0.8.6.

Got it from homebrew, terraform.io needs an update, as the links still point to 0.8.5.
